airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/composite_error_handler.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import List, Union
+
+import requests
+from airbyte_cdk.sources.declarative.requesters.error_handlers.error_handler import ErrorHandler, ResponseAction, ResponseStatus
+
+
+class CompositeErrorHandler(ErrorHandler):
+    """
+    Sample config chaining 2 different retriers:
+        error_handler:
+          type: "CompositeErrorHandler"
+          error_handlers:
+            - response_filters:
+                - predicate: "{{ 'codase' in decoded_response }}"
+                  action: RETRY
+              backoff_strategies:
+                - type: "ConstantBackoffStrategy"
+                  backoff_time_in_seconds: 5
+            - response_filters:
+                - http_codes: [ 403 ]
+                  action: RETRY
+              backoff_strategies:
+                - type: "ConstantBackoffStrategy"
+                  backoff_time_in_seconds: 10
+    """
+
+    def __init__(self, error_handlers: List[ErrorHandler]):
+        """
+
+        :param error_handlers: list of error handlers
+        """
+        self._error_handlers = error_handlers
+        assert self._error_handlers
+
+    @property
+    def max_retries(self) -> Union[int, None]:
+        return self._error_handlers[0].max_retries
+
+    def should_retry(self, response: requests.Response) -> ResponseStatus:
+        retry = None
+        ignore = False
+        for retrier in self._error_handlers:
+            should_retry = retrier.should_retry(response)
+            if should_retry.action == ResponseAction.SUCCESS:
+                return ResponseStatus.success()
+            if should_retry == ResponseStatus.ignore():
+                ignore = True
+            elif retry is None or retry.action != ResponseAction.RETRY:
+                retry = should_retry
+        if ignore:
+            return ResponseStatus.ignore()
```

Comment:
that's a very good point. i'll update to process them sequentially as that's more intuitive.

I don't think it's too restricting because if we wanted the request to be ignored in your example, we could just move the second handler to the head of the list